crypto/src/main/scala/org/bitcoins/crypto/CryptoRuntime.scala

```diff
@@ -136,4 +139,77 @@ trait CryptoRuntime {
   def sha256SchnorrNonce(bytes: ByteVector): Sha256Digest = {
     sha256(schnorrNonceTagBytes ++ bytes)
   }
+
+  def publicKey(privateKey: ECPrivateKey): ECPublicKey
+
+  def sign(privateKey: ECPrivateKey, dataToSign: ByteVector): ECDigitalSignature
+
+  def signWithEntropy(
+      privateKey: ECPrivateKey,
+      bytes: ByteVector,
+      entropy: ByteVector): ECDigitalSignature
+
+  def secKeyVerify(privateKeybytes: ByteVector): Boolean
+
+  def verify(
+      publicKey: ECPublicKey,
+      data: ByteVector,
+      signature: ECDigitalSignature): Boolean
+
+  def decompressed(publicKey: ECPublicKey): ECPublicKey
+
+  def tweakMultiply(publicKey: ECPublicKey, tweak: FieldElement): ECPublicKey
+
+  def add(pk1: ECPrivateKey, pk2: ECPrivateKey): ECPrivateKey
+
+  def add(bytes: ByteVector, pk2: ECPrivateKey): ByteVector
+
+  def add(pk1: ECPublicKey, pk2: ECPublicKey): ECPublicKey
+
+  def pubKeyTweakAdd(pubkey: ECPublicKey, privkey: ECPrivateKey): ECPublicKey
+
+  def isValidPubKey(bytes: ByteVector): Boolean
+
+  def isFullyValidWithBouncyCastle(bytes: ByteVector): Boolean
```

Comment:
there is going not going to be able to reside here as it is bouncy castle specific

Reply:
It must be implemented for all `CryptoRuntime` anyway.

Reply:
probably deserves a rename